stac_fastapi/pgstac/models/links.py

```diff
@@ -51,7 +51,7 @@ def base_url(self):
     @property
     def url(self):
         """Get the current request url."""
-        return str(self.request.url)
+        return urljoin(str(self.request.base_url), str(self.request.url.path).lstrip("/"))
```

Comment:
ok I think we're good with this now 

`self.base_url` would return `http://0.0.0.0/{root-path}/{router-prefix}/collection`  while `self.url.path` returns `{/router-prefix}/collections` so using `self.request.base_url` gives the correct result 

`self.request.url` will return `http://0.0.0.0/{root-path}/collection` because it doesn't know the router-prefix (weird configuration in stac-fastapi it self, because we don't mount the router with the prefix set 🤷)

Reply:
ahhh this won't fix everything 😭 

we also need the query-param part in `self.url`